packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -304,6 +295,58 @@ class LambdaFunctionHotswapOperation implements HotswapOperation {
     // run all of our updates in parallel
     return Promise.all(operations);
   }
+
+  /**
+   * After a Lambda Function is updated, it cannot be updated again until the
+   * `State=Active` and the `LastUpdateStatus=Successful`.
+   *
+   * Depending on the configuration of the Lambda Function this could happen relatively quickly
+   * or very slowly. For example, Zip based functions _not_ in a VPC can take ~1 second whereas VPC
+   * or Container functions can take ~25 seconds (and 'idle' VPC functions can take minutes).
+   */
+  private async waitForLastUpdateStatusComplete(currentFunctionConfiguration: AWS.Lambda.FunctionConfiguration, lambda: AWS.Lambda): Promise<void> {
+    let delay = 1; // in seconds
+    // if the function is deployed in a VPC or if it is a container image function
+    // then the update will take much longer and we can wait longer between checks
+    if (currentFunctionConfiguration.VpcConfig || currentFunctionConfiguration.PackageType === 'Image') {
+      delay = 5;
+    }
```

Comment:
Let's just do it in one expression - I just hate using `let` when a `const` would do:

```suggestion
    const delaySeconds = currentFunctionConfiguration.VpcConfig ||
        currentFunctionConfiguration.PackageType === 'Image'
      // if the function is deployed in a VPC or if it is a container image function
      // then the update will take much longer and we can wait longer between checks
      ? 5
      // otherwise, the update will be quick, so a 1-second delay is fine
      : 1;
```

Reply:
Updated. I also discovered that Lambdas not in a VPC can return a VPC config like

```ts
VpcConfig: { VpcId: '', SubnetIds: [], SecurityGroupIds: [] }
```